lib/online_order.rb

```diff
@@ -0,0 +1,90 @@
+require "csv"
+require "ap"
+require_relative "order.rb"
+require_relative "customer.rb"
+
+module Grocery
+  class OnlineOrder < Order
+    attr_accessor :customer, :online_order_id, :products, :fulfillment_status
+
+      @customer = Customer.find(id)
+      @products = products
```

Comment:
Looks like you're missing a line here. Adding

```ruby
def initialize(online_order_id, products, id, fulfillment_status)
```

between lines 9 and 10 made everything work out.